File: src/main/java/org/apache/fineract/client/services/GuarantorsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.FormDataContentDisposition;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface GuarantorsApi {
  /**
   * 
   * 
   * @param loanId  (required)
   * @param clientId  (optional)
   * @return Observable&lt;String&gt;
   */
  @GET("loans/{loanId}/guarantors/accounts/template")
  Observable<String> accountsTemplate(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Query("clientId") Long clientId
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("loans/{loanId}/guarantors")
  Observable<String> createGuarantor(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param guarantorId  (required)
   * @param guarantorFundingId  (optional)
   * @return Observable&lt;String&gt;
   */
  @DELETE("loans/{loanId}/guarantors/{guarantorId}")
  Observable<String> deleteGuarantor(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Path("guarantorId") Long guarantorId, @retrofit2.http.Query("guarantorFundingId") Long guarantorFundingId
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param officeId  (optional)
   * @param dateFormat  (optional)
   * @return Observable&lt;Void&gt;
   */
  @GET("loans/{loanId}/guarantors/downloadtemplate")
  Observable<Void> getGuarantorTemplate(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Query("officeId") Long officeId, @retrofit2.http.Query("dateFormat") String dateFormat
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("loans/{loanId}/guarantors/template")
  Observable<String> newGuarantorTemplate(
    @retrofit2.http.Path("loanId") Long loanId
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param file  (optional)
   * @param locale  (optional)
   * @param dateFormat  (optional)
   * @return Observable&lt;String&gt;
   */
  @retrofit2.http.Multipart
  @POST("loans/{loanId}/guarantors/uploadtemplate")
  Observable<String> postGuarantorTemplate(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Part("file") FormDataContentDisposition file, @retrofit2.http.Part("locale") String locale, @retrofit2.http.Part("dateFormat") String dateFormat
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("loans/{loanId}/guarantors")
  Observable<String> retrieveGuarantorDetails(
    @retrofit2.http.Path("loanId") Long loanId
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param guarantorId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("loans/{loanId}/guarantors/{guarantorId}")
  Observable<String> retrieveGuarantorDetails1(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Path("guarantorId") Long guarantorId
  );

  /**
   * 
   * 
   * @param loanId  (required)
   * @param guarantorId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("loans/{loanId}/guarantors/{guarantorId}")
  Observable<String> updateGuarantor(
    @retrofit2.http.Path("loanId") Long loanId, @retrofit2.http.Path("guarantorId") Long guarantorId, @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/GuarantorsApi.kt
package org.openapitools.client.apis


import okhttp3.MultipartBody
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path
import retrofit2.http.Query

interface GuarantorsApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param clientId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/loans/{loanId}/guarantors/accounts/template")
    suspend fun accountsTemplate(
        @Path("loanId") loanId: Long,
        @Query("clientId") clientId: Long? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/loans/{loanId}/guarantors")
    suspend fun createGuarantor(
        @Path("loanId") loanId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param guarantorId
     * @param guarantorFundingId  (optional)
     * @return [kotlin.String]
     */
    @DELETE("v1/loans/{loanId}/guarantors/{guarantorId}")
    suspend fun deleteGuarantor(
        @Path("loanId") loanId: Long,
        @Path("guarantorId") guarantorId: Long,
        @Query("guarantorFundingId") guarantorFundingId: Long? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param officeId  (optional)
     * @param dateFormat  (optional)
     * @return [Unit]
     */
    @GET("v1/loans/{loanId}/guarantors/downloadtemplate")
    suspend fun getGuarantorTemplate(
        @Path("loanId") loanId: Long,
        @Query("officeId") officeId: Long? = null,
        @Query("dateFormat") dateFormat: String? = null
    ): Unit

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @return [kotlin.String]
     */
    @GET("v1/loans/{loanId}/guarantors/template")
    suspend fun newGuarantorTemplate(@Path("loanId") loanId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param dateFormat  (optional)
     * @param locale  (optional)
     * @param uploadedInputStream  (optional)
     * @return [kotlin.String]
     */
    @Multipart
    @POST("v1/loans/{loanId}/guarantors/uploadtemplate")
    suspend fun postGuarantorTemplate(
        @Path("loanId") loanId: Long,
        @Part("dateFormat") dateFormat: String? = null,
        @Part("locale") locale: String? = null,
        @Part uploadedInputStream: MultipartBody.Part? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @return [kotlin.String]
     */
    @GET("v1/loans/{loanId}/guarantors")
    suspend fun retrieveGuarantorDetails(@Path("loanId") loanId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param guarantorId
     * @return [kotlin.String]
     */
    @GET("v1/loans/{loanId}/guarantors/{guarantorId}")
    suspend fun retrieveGuarantorDetails1(
        @Path("loanId") loanId: Long,
        @Path("guarantorId") guarantorId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanId
     * @param guarantorId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/loans/{loanId}/guarantors/{guarantorId}")
    suspend fun updateGuarantor(
        @Path("loanId") loanId: Long,
        @Path("guarantorId") guarantorId: Long,
        @Body body: String? = null
    ): String

}
